build/tools/FileSystem.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.IO;
+using Nuke.Common.IO;
+
+namespace Tools
+{
+    public static class FileSystem
+    {
+        public static string[] GetFiles(AbsolutePath path, string filter)
+        {
+            if (!Directory.Exists(path))
+            {
+                Directory.CreateDirectory(path!);
```

Comment:
Creating a directory from a `GetFiles` call would be an unexpected side-effect. Consider throwing an exception instead if the directory is not found instead of silently pretending the caller sent something valid.

Also, if `AbsolutePath` can be null here(?), consider checking for null and throw an `ArgumentNullException`.  The null-forgiving operator (`!`) should be an absolute last resort.

Reply:
And done